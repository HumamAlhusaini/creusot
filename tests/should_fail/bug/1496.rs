extern crate creusot_contracts;

pub fn foo<T>(x: *const T) -> &'static T {
    unsafe { &*x }
}
