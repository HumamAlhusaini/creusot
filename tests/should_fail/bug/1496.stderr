error: Dereference of a raw pointer is forbidden in creusot: use `creusot_contracts::ptr_own::PtrOwn` instead
 --> 1496.rs:4:14
  |
4 |     unsafe { &*x }
  |              ^^^

error: aborting due to 1 previous error

